pkg/cloud/cloudcheck/cloudcheck_stmt.go

```diff
@@ -0,0 +1,100 @@
+// Copyright 2023 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package cloudcheck
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/cockroachdb/cockroach/pkg/cloud/cloudprivilege"
+	"github.com/cockroachdb/cockroach/pkg/sql"
+	"github.com/cockroachdb/cockroach/pkg/sql/catalog/colinfo"
+	"github.com/cockroachdb/cockroach/pkg/sql/execinfrapb"
+	"github.com/cockroachdb/cockroach/pkg/sql/physicalplan"
+	"github.com/cockroachdb/cockroach/pkg/sql/sem/tree"
+	"github.com/cockroachdb/cockroach/pkg/sql/types"
+	"github.com/cockroachdb/cockroach/pkg/util/tracing"
+	"github.com/pkg/errors"
+)
+
+var Header = colinfo.ResultColumns{
+	{Name: "node", Typ: types.Int},
+	{Name: "locality", Typ: types.String},
+	{Name: "ok", Typ: types.Bool},
+	{Name: "error", Typ: types.String},
+	{Name: "read_speed", Typ: types.String},
+	{Name: "write_speed", Typ: types.String},
+	{Name: "can_delete", Typ: types.Bool},
+}
+
+// ShowCloudStorageTestPlanHook is currently called by showBackup hook but
+// should be extended to be a standalone plan instead.
+func ShowCloudStorageTestPlanHook(
+	ctx context.Context, p sql.PlanHookState, location string,
+) (sql.PlanHookRowFn, colinfo.ResultColumns, []sql.PlanNode, bool, error) {
+
+	if err := cloudprivilege.CheckDestinationPrivileges(ctx, p, []string{location}); err != nil {
+		return nil, nil, nil, false, err
+	}
+
+	fn := func(ctx context.Context, _ []sql.PlanNode, resultsCh chan<- tree.Datums) error {
```

Comment:
Maybe print a warning about how experimental this feature is and how it is subject to change without warning at any time in any release?

Reply:
also maybe a warning that this will degrade external IO while it's running